lib/index.js

```diff
@@ -29,7 +31,16 @@ function type(V) {
             // Per ES spec, typeof returns an implemention-defined value that is not any of the existing ones for
             // uncallable non-standard exotic objects. Yet Type() which the Web IDL spec depends on returns Object for
             // such cases. So treat the default case as an object.
-            return "Object";
+
+            // This will also be hit for objects with the `[[IsHTMLDDA]]` internal slot,
+            // which causes typeof to return "undefined".
+            try {
+                // %ObjProto_valueOf% acts an identity function for real objects
+                if (objValueOf.call(V) === V) {
+                    return "Object";
+                }
+            } catch (err) { /* Ignore */ }
+            return "Unknown";
```

Comment:
It seems like we don't have to add this try-catch block, given that we have tightened up the check for Undefined values up above. "Unknown" is not a ECMA-262 value type, and we can assume that our host implements ECMA-262.

Reply:
This is a fallback for any potential future primitive type, though I guess I could do what ES‑Abstract does and return `undefined` in this case.